stag-library/src/main/java/com/vimeo/stag/UseStag.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * The MIT License (MIT)
+ * <p/>
+ * Copyright (c) 2016 Vimeo
+ * <p/>
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * <p/>
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ * <p/>
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.vimeo.stag;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Target;
+
+/**
+ * Use this annotation to tell Stag that all the member variables of the class should be processed.
+ * <p/>
+ * If the class is annotated, Stag will generate a TypeAdapter for that class.
+ * Stag does not generate TypeAdapters for abstract classes.
+ * <p/>
+ */
+@Target({ElementType.TYPE})
+public @interface UseStag {
```

Comment:
I hadn't given thought to that GsonBuilder method (or even the exclude with modifiers one), you're right, just relying on transient would be a problem. Maybe we should introduce our own "excludeFieldsWith...?"

I also forgot about the use case for `@GsonAdapterKey` where you can annotate a class without any fields (the example in the model being `VideoList`). In that case we don't want any direct fields of that class touched by stag, but we want all the inherited fields to be resolved and parsed/serialized. `@UseStag` currently would just grab all the fields rather than just tagging the class for stag, which would be problematic.

As to your last comment, that seems like a possibility - require either `@GsonAdapterKey` + `@SerializedName` usage, or `@UseStag` + `@SerializedName`. My fear with enforcing this is that it would become very confusing for consumers of the library as the nuances between the use of the two annotations are confusing without in depth use of the library.

Another option would be for `@UseStag` to take on the responsibilities of `@GsonAdapterKey`. When applied at the class level, it would specify whether stag should pick up all fields or look for annotated fields.

```java
// Use case 1
@UseStag(options = Options.ALL)
public class Model {
    transient String ignored;

    String string1;

    @SerializedName("stringy")
    String string2;

    @UseStag("strings")
    List<String> stringList;
}

// Use case 2
@UseStag(options = Options.NONE)
public class Model2 extends Model3<String> {
    boolean initializationFlag;

    // If this field exists, we'll pick it up.
    // If it doesn't exist, we'll just generate a type adapter based on the inherited, resolved fields.
    @SerializedName("optional")
    boolean optionalThing;
}

public class Model3<T> {
    @UseStag
    T data;
}

// Use case 3
public class Model4 {
    String notSerialized;

    @UseStag
    String serialized;

    @UseStag("serialized_data")
    String serialized2;
}
```

Or we could use `@GsonAdapterKey` instead of the `@UseStag` annotation since that's what consumers are already using. However, since this will be for a 2.0 release, a name change shouldn't be a problem.

cc @kvenn @kevinzetterstrom